packages/ai-jsx/src/lib/llama2.tsx

```diff
@@ -0,0 +1,174 @@
+    prompt,
+    max_length: props.maxTokens ?? defaultMaxTokens,
+    repetition_penalty: props.repetitionPenalty,
+    temperature: props.temperature,
+    top_p: props.topP,
+  };
+  logger.debug({ llama2Args }, 'Calling Llama2');
+  const response = await props.fetch(llama2Args);
+  yield* response;
+  return AI.AppendOnlyStream;
+}
+
+async function* Llama2CompletionModel(
+  props: Llama2Props,
+  { render, logger }: AI.ComponentContext
+): AI.RenderableStream {
+  yield AI.AppendOnlyStream;
+  const prompt = await render(props.children);
+  const llama2Args: FetchLlama2Args = {
+    prompt,
+    max_length: props.maxTokens ?? defaultMaxTokens,
+    temperature: props.temperature,
+    top_p: props.topP,
+  };
+  logger.debug({ llama2Args }, 'Calling Llama2');
+  const response = await props.fetch(llama2Args);
+  yield* response;
+  return AI.AppendOnlyStream;
+}
+
+/**
+ * This component causes all children `ChatCompletion` and `Completion` components to use Llama2.
+ *
+ * Because Llama2 is open source, there's no single model provider like there is for OpenAI and Anthropic.
+ * If you're using this component, you probably have your own Llama2 instance hosted somewhere.
+ * To point this component to your Llama2 instance, pass a `fetchLlama2` function that returns a generator that yields chunks of the response.
+ *
+ * @see ReplicateLlama2 if you have a model hosted on Replicate.
+ * @see Llama2ChatModel for notes about polyfilling.
+ */
+export function Llama2({
+  children,
+  fetchLlama2,
+  ...defaults
+}: { children: Node; fetchLlama2: FetchLlama2 } & Llama2ModelProps) {
+  return (
+    <ChatProvider component={Llama2ChatModel} {...defaults} fetch={fetchLlama2}>
+      <CompletionProvider component={Llama2CompletionModel} {...defaults} fetch={fetchLlama2}>
+        {children}
+      </CompletionProvider>
+    </ChatProvider>
+  );
+}
+
+/**
+ * This component causes all children `ChatCompletion` and `Completion` components to use Llama2 hosted on Replicate.
+ *
+ * You must set env var REPLICATE_API_TOKEN or REACT_REPLICATE_API_TOKEN for this to work.
+ *
+ * Drawbacks to this implementation:
+ *  * The Replicate JS client lib does not support streaming.
+ *  * The Replicate API will sometimes return an empty string with no explanation.
+ *      (This has been observed when passing maxTokens=50.)
+ */
+export function ReplicateLlama2(
+  { children, ...defaults }: { children: Node } & Llama2ModelProps,
+  { logger }: AI.ComponentContext
+) {
+  const replicate = new Replicate({
+    auth: getEnvVar('REPLICATE_API_TOKEN', true)!,
+  });
+
+  async function* fetchLlama2(input: FetchLlama2Args) {
+    const output = (await replicate.run(
+      'replicate/llama70b-v2-chat:e951f18578850b652510200860fc4ea62b3b16fac280f83ff32282f87bbd2e48',
+      { input }
+    )) as string[];
+    const result = output.join('');
+    logger.debug({ output, result }, 'Replicate output');
+    yield result;
+    return result;
```

Comment:
Unbelievably, this doesn't support streaming. The closest it has is the ability to poll an endpoint. 🤮